modules/test/conn/python/src/connection_module.py

```diff
@@ -74,6 +74,60 @@ def __init__(self, module):
     # response = self.dhcp1_client.set_dhcp_range('10.10.10.20','10.10.10.30')
     # print("Set Range: " + str(response))
 
+  def _connection_switch_arp_inspection(self):
+    LOGGER.info('Running connection.switch.arp_inspection')
+
+    no_arp = True
+
+    # Read all the pcap files
+    packets = rdpcap(STARTUP_CAPTURE_FILE) + rdpcap(MONITOR_CAPTURE_FILE)
+    for packet in packets:
+
+      # We are not interested in packets unless they are ARP packets
+      if not ARP in packet:
+        continue
+
+      # We are only interested in packets from the device
+      if packet.src != self._device_mac:
+        continue
+
+      # Get the ARP packet
+      arp_packet = packet[ARP]
+      no_arp = False
+
+      # Check MAC address matches IP address
+      if (arp_packet.hwsrc != self._device_mac or
+          arp_packet.psrc != self._device_ipv4_addr):
+        return False, 'Device is sending false ARP response'

```

Comment:
Logic check needs to be updated so that we are checking if the expected IP is being used for the correct mac.  Currently this filters for any ARP that isn't the device mac or IP which isn't correct for arp insepection.

This fixes that and adds some useful logging for failures:
```
      # Check MAC address matches IP address
      if (arp_packet.hwsrc == self._device_mac and
          arp_packet.psrc != self._device_ipv4_addr):
        LOGGER.info(f'Bad ARP packet detected for MAC: {self._device_mac}')
        LOGGER.info(f'ARP packet IP {arp_packet.psrc} does not match {self._device_ipv4_addr}')
        return False, 'Device is sending false ARP response'
```